.changeset/stupid-foxes-type.md

```diff
@@ -0,0 +1,20 @@
+---
+'@backstage/plugin-catalog-react': patch
+---
+
+Use the history API directly in `useEntityListProvider`.
+
+This replaces `useSearchParams`/`useNavigate`, since I found that they caused additional rerenders.
+
+Table re-render count is down additionally:
+
+- Initial render of catalog page: 12 -> 9
+- Full `CatalogPage` test: 57 -> 48
+
+Also noting that `react-table` warns at first render:
+
+```
+The columns provided to material table are static, but contain functions which update on every render, resetting the table state. Provide a stable function or column reference or an row id to prevent state loss.
+```
+
+I experimented quickly around that too but didn't make any progress, will have to postpone until some other time.
```

Comment:
⭐ for looking at reducing the re-renders. I can't say direct history manipulation is right or wrong but I got to ask about the changeset. Do we want the scribbles of a mad scientist in the changelog? 😂

Reply:
haha that's actually a good callout, i'll move it to the pr description :D :D